modules/system-view/src/main/java/org/apache/ignite/internal/systemview/SystemViewManager.java

```diff
@@ -32,4 +36,22 @@ public interface SystemViewManager extends IgniteComponent {
      * @param view System view to register.
      */
     void register(SystemView<?> view);
+
+    /**
+     * Returns a list of nodes a view with given name can be found on.
+     *
+     * @param name Name of view of interest.
+     * @return A list of nodes owning a view. Never empty.
```

Comment:
How we could achive it in case all nodes with the view leave a cluster? I think we just should return emty result for query which try to select fro such views.

Reply:
as it stated on the very next line, exception will be thrown in that case

Reply:
changed this a bit: now manager returns empty list if no nodes were found